test/bin/common.sh

```diff
@@ -268,3 +271,92 @@ get_scenario_type_from_path() {
     esac
     echo "${type}"
 }
+
+# Prepare a static delta file between two container images specified in the
+# function arguments. The images are copied from the mirror registry to a local
+# directory and compared using the 'tar-diff' command.
+#
+# The result is stored in the '${BOOTC_STATIC_DELTA}/${src}@${dst}.tardiff' file.
+#
+# Arguments:
+#   src - Source container image name
+#   dst - Destination container image name
+# Return:
+#   '${src}@${dst}.tardiff' file stored in the '${BOOTC_STATIC_DELTA}' directory
+prepare_static_delta() {
+    local -r src=$1
+    local -r dst=$2
+    local -r tardiff="${src}@${dst}.tardiff"
+
+    local -r workdir="$(mktemp -d /tmp/bootc-delta-prepare.XXXXXXXX)"
+    # shellcheck disable=SC2164
+    pushd "${workdir}" &>/dev/null
+
+    "${ROOTDIR}/scripts/fetch_tools.sh" tar-diff
+
+    # Export the images and archive them
+    # Note: Cannot use 'docker-archive' because it only supports Docker Schema 2 manifests
+    for i in "${src}" "${dst}" ; do
+        skopeo copy --all --preserve-digests \
+            --authfile "${MIRROR_REGISTRY_DIR}/config/microshift_auth.json" \
+            docker://"${MIRROR_REGISTRY_URL}/${i}:latest" \
+            oci-archive:"${i}.tar"
+    done
+
+    # Create a tar-diff file and list tar sizes
+    "${ROOTDIR}/_output/bin/tar-diff" "${src}.tar" "${dst}.tar" "${tardiff}"
+    ls -lh ./*.tar*
+
+    # Store the tar-diff file in the static delta directory
+    mkdir -p "${BOOTC_STATIC_DELTA}"
+    mv -f "${tardiff}" "${BOOTC_STATIC_DELTA}/"
+
+    # shellcheck disable=SC2164
+    popd &>/dev/null
+    rm -rf "${workdir}"
+}
+
+# Consume a static delta file created by the 'prepare_static_delta' function.
+# The '${src}' image is copied from the mirror registry to a local directory
+# and merged with the '${BOOTC_STATIC_DELTA}/${src}@${dst}.tardiff' file.
+#
+# The result of the merge is uploaded to the mirror registry and stored under
+# the '${dst}-patched' name.
+#
+# Arguments:
+#   src - Source container image name
+#   dst - Destination container image name
+# Return:
+#   '${dst}-patched' image stored in the mirror registry
+consume_static_delta() {
```

Comment:
Nit: I find the 'consume' word a bit non descriptive and vague - it doesn't hint anything about what happens or what's involved in the process.
Although mouthful, I would find something like 'merge static detal with src' a bit more obvious at first glance.
We don't have to change it just now, but I would appreciate clarifying it some time.